ast-nodes/declaration.dart

```diff
@@ -1,10 +1,13 @@
 import 'index.dart';
 import '../lexer/token.dart';
 import '../errors/index.dart';
+import 'dart:ffi';
+import '../codegen/index.dart';
 
 /// A declaration is a [Statement] that creates a new entity with a [name].
 abstract class Declaration implements Statement {
   String name;
+  Pointer<LLVMOpaqueValue> opaqueValue;
```

Comment:
`opaqueValue` is a little extreme for a name, it's only opaque in the sense that Dart doesn't see what's going on in the C world. Maybe `valueRef`?